apps/varnish.go

```diff
@@ -52,3 +53,29 @@ func (r MetricsReceiverVarnish) Pipelines() []otel.Pipeline {
 func init() {
 	confgenerator.MetricsReceiverTypes.RegisterType(func() confgenerator.Component { return &MetricsReceiverVarnish{} })
 }
+
+type LoggingProcessorVarnish struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorVarnish) Type() string {
+	return "varnish"
+}
+
+func (p LoggingProcessorVarnish) Components(tag string, uid string) []fluentbit.Component {
+	c := confgenerator.LoggingProcessorParseRegex{
+		// Logging documentation: https://github.com/varnishcache/varnish-cache/blob/04455d6c3d8b2d810007239cb1cb2b740d7ec8ab/doc/sphinx/reference/varnishncsa.rst#format
+		// Sample line: 127.0.0.1 - - [02/Mar/2022:15:55:05 +0000] "GET http://localhost:8080/test HTTP/1.1" 404 273 "-" "curl/7.64.0"
+		Regex: `^(?<http_request_serverIp>[^ ]*) ([^ ]*) (?<http_request_remoteIp>[^ ]*) \[(?<time>[^\]]*)\] "(?<http_request_requestMethod>\S+)(?: +(?<http_request_requestUrl>[^\"]*?)(?: +(?<http_request_protocol>\S+))?)?" (?<http_request_status>[^ ]*) (?<http_request_responseSize>[^ ]*)(?: "(?<http_request_referer>[^\"]*)" "(?<http_request_userAgent>[^\"]*)")?$`,
```

Comment:
For the user to use this plugin, what will they have to do to set up Varnish? It looks like we can ask them to use a specific logging format (https://docs.varnish-software.com/tutorials/enabling-logging-with-varnishncsa/#step-3-customize-options-1) and then reuse that here.

Reply:
To minimize user configuration we were leaving the varnishncsa logging format as default, is there something valuable you're seeing in the custom log format that appears to be worth the extra configuration requirement?

Reply:
Ack. Not a whole lot of benefit, given that this configuration lives outside the purview of the Ops Agent, it feels natural to me that we provide a log option pattern that the Ops Agent expects. If the user configures the logs to be structured differently, their Ops Agent logs will become unstructured and that user surprise is what I was hoping to minimize. 

But our agent could also just document the configuration requirements that it requires from Varnish to work as intended and that should be enough.